client/src/components/LibraryNavigator/index.ts

```diff
@@ -29,7 +31,7 @@ class LibraryNavigator implements SubscriptionProvider {
   constructor(context: ExtensionContext) {
     this.extensionUri = context.extensionUri;
     this.libraryDataProvider = new LibraryDataProvider(
-      new LibraryModel(),
+      new LibraryModel(this.activeProfileConnectionType()),
```

Comment:
nit: just wondering is it a good idea to have LibraryModel taking care of the active connection type internally?

Reply:
Not a bad idea. So maybe instead of `LibraryDataProvider.refresh(connectionType: ConnectionType)`, we have something more like `LibraryDataProvider.useAdapter(adapter: LibraryAdapter)` that gets passed down to the model?

Reply:
Resolved in https://github.com/sassoftware/vscode-sas-extension/pull/733/commits/93d3eff37b0ed5e1302e7be4f63a9cbe3a4ac847